Prog/Affordability_sf_condo.sas

```diff
@@ -0,0 +1,321 @@
+/**************************************************************************
+Program:  Affordability_sf_condo.sas
+ Library:  DMPED
+ Project:  NeighborhoodInfo DC
+ Author:   Yipeng Su
+ Created:  6/28/18
+ Version:  SAS 9.4
+ Environment:  Windows with SAS/Connect
+ 
+ Description: *Methodology for affordability adapted from Zhong Yi Tong paper 
+http://content.knowledgeplex.org/kp2/cache/documents/22736.pdf
+Homeownership Affordability in Urban America: Past and Future;
+ Modifications: 
+**************************************************************************/
+
+
+%include "L:\SAS\Inc\StdLocal.sas";
+
+
+** Define libraries **;
+%DCData_lib( DMPED );
+%DCData_lib( equity );
+%DCData_lib( realprop );
+
+%Sales_pars();
+%Clean_sales( inds=DMPED.Sales_who_owns_SF_Condo, outds=Sales_who_owns_SF_Condo_clean) ;
+
+proc contents data= Sales_who_owns_SF_Condo_clean; run;
+
+data create_flags;
+  set Sales_who_owns_SF_Condo_clean (where=(LargeUnit=1 and clean_sale=1 and (2000 <= saleyear <= 2017)));
+  
+  /*pull in effective interest rates - for example: 
+  http://www.fhfa.gov/DataTools/Downloads/Documents/Historical-Summary-Tables/Table15_2015_by_State_and_Year.xls*/
+  
+    eff_int_rate_2000= 7.90;
+	eff_int_rate_2001= 7.07;
+	eff_int_rate_2002= 6.62;
+	eff_int_rate_2003= 5.73;
+	eff_int_rate_2004= 5.53;
+	eff_int_rate_2005= 5.67;
+	eff_int_rate_2006= 6.53;
+	eff_int_rate_2007= 6.54;
+	eff_int_rate_2008= 6.14;
+	eff_int_rate_2009= 5.02;
+    eff_int_rate_2010= 4.93;
+	eff_int_rate_2011= 4.62;
+	eff_int_rate_2012= 3.72;
+	eff_int_rate_2013= 3.95;
+	eff_int_rate_2014= 4.22;
+	eff_int_rate_2015= 3.95;
+	eff_int_rate_2016= 3.69;
+    eff_int_rate_2017= 3.69; *2017 not available, using 2016;
+	%macro yearloop ();
+
+	%do year = 2000 %to 2017 ;
+
+	month_int_rate_&year. = (eff_int_rate_&year./12/100);
+
+	loan_multiplier_&year. =  month_int_rate_&year. *	( ( 1 + month_int_rate_&year. )**360	) / ( ( ( 1+ month_int_rate_&year. )**360 )-1 );
+
+
+	*calculate monthly Principal and Interest for First time Homebuyer (10% down);
+	if saleyear=&year. then PI_First&year.=saleprice*.9*loan_multiplier_&year.;
+
+	%dollar_convert(PI_first&year.,PI_first&year.r,&year.,2017);
```

Comment:
@maggiesu0725 - take out the code for inflation adjustment - the incomes are not inflation adjusted so we shouldn't do the prices.

Reply:
take out code for inflation adjustment